manager/src/main/java/com/redhat/service/bridge/manager/ProcessorServiceImpl.java

```diff
@@ -136,6 +137,51 @@ public Processor createProcessor(String bridgeId, String customerId, ProcessorRe
         return newProcessor;
     }
 
+    @Override
+    @Transactional
+    public Processor updateProcessor(String processorId, String bridgeId, String customerId, ProcessorRequest processorRequest) {
+        // Attempt to load the Bridge. We cannot update a Processor if the Bridge is not available.
+        bridgesService.getReadyBridge(bridgeId, customerId);
+
+        // Extract existing definition
+        Processor existingProcessor = getProcessor(processorId, bridgeId, customerId);
+        ProcessorDefinition existingDefinition = jsonNodeToDefinition(existingProcessor.getDefinition());
+        String existingTransformationTemplate = existingDefinition.getTransformationTemplate();
+        BaseAction existingAction = existingDefinition.getRequestedAction();
+        BaseAction existingResolvedAction = existingDefinition.getResolvedAction();
+
+        // Validate update. Only Filters are supported at present.
+        if (!Objects.equals(existingProcessor.getName(), processorRequest.getName())) {
+            throw new BadRequestException("It is not possible to update the Processor's name.");
+        }
+        if (!Objects.equals(existingTransformationTemplate, processorRequest.getTransformationTemplate())) {
+            throw new BadRequestException("It is not possible to update the Processor's TransformTemplate.");
+        }
+        if (!Objects.equals(existingAction, processorRequest.getAction())) {
+            throw new BadRequestException("It is not possible to update the Processor's Action.");
+        }
+
+        // Create new definition copying existing properties
+        existingProcessor.setSubmittedAt(ZonedDateTime.now());
```

Comment:
@manstis Should we be changing `modifiedAt` rather than `submittedAt`? Otherwise the customer loses record of when this `Processor` was originally created.

cheers,

Rob

Reply:
Looking at kafka they have `created_at` as well as `updated_at`. Same for connectors, but they have `created_at` and `modified_at`. 

I'd stay with kafka

Reply:
Hmm @robpblake Our `Worker` mechanism [times-out](https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/blob/main/manager/src/main/java/com/redhat/service/bridge/manager/workers/resources/AbstractWorker.java#L125) `Work` if it is excessively "old" (from the submitted date). Therefore, if we want to keep the preserve the original `submittedAt` and update the `modifiedAt` when updating a `Processor` we should change the `Worker` mechanism too. 

Let me know your thoughts on use of `submittedAt` for work expiration and I will create another JIRA if we need to change both this (updates set the `modifiedAt` and work expires based on `modifiedAt` too.... but TBH it may open a can of undesirable worms as we [changed the timeout to be based on the `submittedAt`](https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/pull/402#discussion_r818904373) when the `Worker` mechanism was first written).

Reply:
If we don't update the `submittedAt` the new `Work` to update the `Processor` is likely to be marked as "timed out" before it even executes... so, as it stands, we have no choice other than to update it.

Reply:
@manstis Correct me if I'm wrong, but the timeout for our `Worker` is based on `Work.getSubmittedAt` and not the `ManagedResource.getSubmittedAt`?

Reply:
Damn you Mr Blake!

You are (probably) correct... let me double check my double checking :smile.

Reply:
Change the call here to `setModified(..)` as @robpblake correctly proposed the first time.